KOGITO-3211 Introducting CI example with Kogito SW, Knative and Camel (#351)
kiegroup/kogito-examples: serverless-workflow-github-showcase/README.md

![](docs/github-showcase-architecture-overview.png)

The implementation relies on three services included in this repository:

1. [**Pull Request Checker Workflow**](pr-checker-workflow): it will coordinate the changes in the PR 
opened in a given GitHub repository. Depending on the files changed in the project,
different labels and reviewers will be automatically added in the PR. In the end of 
the workflow, the service will broadcast a "PR Verified" event to the cluster.
2. [**GitHub API Wrapper Service**](github-service): calling the GitHub API as a GitHub Application
requires a token to be generated on a small timeframe. This service generates a valid
token each 5 minutes to make calls to the API. Besides token management, it wraps 
the API and simplifies its interface just for the sake of this example.
3. [**Notification Service**](notification-service): a simple Camel service to interact with the Slack API to 
notify a given channel. 

### Prerequisites

> **IMPORTANT!** Before proceeding, please make sure you have everything listed in this section ready.

You may use CRC or Minikube if you don't have a cluster available with cluster admin rights.

> Or you can ask an administrator to install the prereqs for you.

Reviewer: ```suggestion
Or you can ask an administrator to install the prerequesites for you.
```

> To deploy this example in your Kubernetes/OpenShift cluster, you will need:

Reviewer: ```suggestion
To deploy this example in your Kubernetes/OpenShift cluster, you will need (replace `<YOUR_NAMESPACE>` placeholder):
```

Author: I will leave as is because we need that namespace for the example to work.

1. A [Quay.io](https://quay.io/repository/) account

> 2. A Kubernetes/OpenShift namespace to deploy the example: `kubectl create ns kogito-github`

Reviewer: ```suggestion
2. A Kubernetes/OpenShift namespace to deploy the example: `kubectl create ns <YOUR_NAMESPACE>`
```

Author: Actually `kustomize` needs a fixed namespace. I could sed into the file, but this would be simpler, that's why I kept this name constant.

Reviewer: then you should mention it somewhere that it should remain

> 3. [**Istio**](https://istio.io/docs/setup/install/istioctl/) installed. 

Reviewer 2: knative typically mandates Istio (or at least a similar ["networking layer"](https://knative.dev/docs/install/any-kubernetes-cluster/))

You can follow the [Knative documentation](https://knative.dev/development/install/installing-istio) for a very basic and simple installation.
4. **Knative** Serving and Eventing components installed. 
We recommend [installing the Knative Operator](https://knative.dev/docs/install/knative-with-operators/) and install the rest of the components
through it as described in their documentation.

> 5. **Kogito Operator** installed in the namespace `kogito-github`. [Download the latest release](https://github.com/kiegroup/kogito-cloud-operator/releases), and run: `NAMESPACE=kogito-github ./hack/install.sh`

Reviewer: ```suggestion
5. **Kogito Operator** installed in the created namespace. [Download the latest release](https://github.com/kiegroup/kogito-cloud-operator/releases), and run: `NAMESPACE=<YOUR_NAMESPACE> ./hack/install.sh`
```

Author: Same as above

In your local machine you will need:

> 1. To clone this repository: `https://github.com/kiegroup/kogito-examples.git && cd serverless-workflow-github-showcase`

Reviewer: ```suggestion
1. To clone this repository and go to `serverless-workflow-github-showcase` directory (`git clone https://github.com/kiegroup/kogito-examples.git && cd serverless-workflow-github-showcase`)
```
